### ISSUE TITLE
Render fails with Client Error for all RSC .server components

### ISSUE THREAD

**Issue by KlotzJesse** [vercel/next.js#32514]
### What version of Next.js are you using?

12.0.8-canary.5 up to 12.0.8-canary.7

### What version of Node.js are you using?

local: 14.8.1 // vercel: 14

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

Vercel

### Describe the Bug

When updating to newest canaries, starting by 12.0.8-canary.5, using `serverComponents: true` and `concurrentFeatures: true`, every .server page fails with a client error.

Browser:
```Application error: a client-side exception has occurred (see the browser console for more information).```

Console (Minimal Reproduction):
![image](https://user-images.githubusercontent.com/9445140/146084472-67421426-0721-4349-891f-00202ed14249.png)

Console (My Production):
![image](https://user-images.githubusercontent.com/9445140/146084563-9a6d77b7-a3c4-47c3-a75c-9412d4b83919.png)

Vercel Error:
![image](https://user-images.githubusercontent.com/9445140/146085208-135b8f15-7ef8-458b-b66d-22b9fc79ec64.png)

The .client pages runs without problem.

**Locally everything is working as supposed.**

### Expected Behavior

Should render as on 12.0.8-canary.4 without issues

### To Reproduce

Update to canary 12.0.8-canary.5 up to .7
Activate in next.config.js
```
module.exports = {
  experimental: {
    reactRoot: true,
    concurrentFeatures: true,
    serverComponents: true,
  },
}
```

Add a Server Component Page, like test.server.js.
Deploy to Vercel
Run into failure

Minimal Reproduction:
Failure (server): https://next-rsc-demo-e3pu88wfg-klotzjesse.vercel.app/rsc
Running (client): https://next-rsc-demo-e3pu88wfg-klotzjesse.vercel.app

Repo: https://github.com/KlotzJesse/next-rsc-demo/tree/rsc-fails-render-with-client-error

**Comment by KlotzJesse** [comment]
Started working on v12.0.8-canary.10 again, as there is no related fix for this, i think it is related to v12.0.8-canary.9

Can anyone, probably @shuding, confirm on that?
Is this therefore fully resolved or has it to be further investigated?

**Comment by balazsorban44** [comment]
Hi, since Next.js 13, React Server Components is only supported in the `app` directory. See https://beta.nextjs.org/docs/getting-started.

If you still have issues, please open a new issue with a reproduction, using `next@canary`. Thanks. :pray:
